src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -559,7 +559,12 @@ internal MemberAccessor MemberAccessorStrategy
             {
                 if (_memberAccessorStrategy == null)
                 {
-#if NETFRAMEWORK || NETCOREAPP
+#if NETCOREAPP
+                    // if dynamic code isn't supported, fallback to reflection
+                    _memberAccessorStrategy = RuntimeFeature.IsDynamicCodeSupported ?
+                        new ReflectionEmitMemberAccessor() :
+                        new ReflectionMemberAccessor();
+#elif NETFRAMEWORK
```

Comment:
What's the else block for... netstandard2.0?  I'm curious why that one differs from NETFRAMEWORK?

Reply:
Correct, `netstandard2.0`.

https://github.com/dotnet/runtime/blob/643181988be5429bea709f0e689b47b642a8b12a/src/libraries/System.Text.Json/src/System.Text.Json.csproj#L4

I was actually just testing this stuff on Xamarin iOS 14 (the current "in market" Xamarin bits). And the reason JsonSerializer works on iOS there is because it picks the `netstandard2.0` assembly, which uses `ReflectionMemberAccessor`.